plugin-k8s/src/k8s_probe.rs

```diff
@@ -52,57 +66,115 @@ impl alumet::pipeline::Source for K8SProbe {
         timestamp: Timestamp,
     ) -> Result<(), alumet::pipeline::PollError> {
         let mut file_buffer = String::new();
+        let mut element_unreachable: Vec<String> = Vec::new();
         for (metric_file, counter_tot, counter_usr, counter_sys) in &mut self.metric_and_counter {
-            let metrics: CgroupV2Metric = cgroup_v2::gather_value(metric_file, &mut file_buffer)?;
-            let diff_tot = match counter_tot.update(metrics.time_used_tot) {
-                CounterDiffUpdate::FirstTime => None,
-                CounterDiffUpdate::Difference(diff) | CounterDiffUpdate::CorrectedDifference(diff) => Some(diff),
+            match cgroup_v2::gather_value(metric_file, &mut file_buffer){
+                Ok(metrics_gathered) => {
+                    let diff_tot = match counter_tot.update(metrics_gathered.time_used_tot) {
+                        CounterDiffUpdate::FirstTime => None,
+                        CounterDiffUpdate::Difference(diff) | CounterDiffUpdate::CorrectedDifference(diff) => Some(diff),
+                    };
+                    let diff_usr = match counter_usr.update(metrics_gathered.time_used_user_mode) {
+                        CounterDiffUpdate::FirstTime => None,
+                        CounterDiffUpdate::Difference(diff) => Some(diff),
+                        CounterDiffUpdate::CorrectedDifference(diff) => Some(diff),
+                    };
```

Comment:
I notice this pattern a lot (in other plugins too). Do you think that it would be a good thing to add a method `difference` to the `alumet` crate, which would turn your code into the following?
```rs
let diff_usr = counter_usr.update(metrics_gathered.time_used_user_mode).difference();
```